src/app/icd/client/CheckInHandler.h

```diff
@@ -33,19 +35,27 @@
 
 namespace chip {
 namespace app {
-class CheckInMessageHandler : public Messaging::ExchangeDelegate, public Messaging::UnsolicitedMessageHandler
+
+class CheckInHandler : public Messaging::ExchangeDelegate, public Messaging::UnsolicitedMessageHandler
 {
 
 public:
     CHIP_ERROR Init(Messaging::ExchangeManager * exchangeManager, ICDClientStorage * clientStorage, CheckInDelegate * delegate);
     void Shutdown();
 
-    virtual ~CheckInMessageHandler() = default;
+    CheckInHandler();
+
+    virtual ~CheckInHandler() = default;
+
+    static void HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr,
```

Comment:
please add doxygen for HandleDeviceConnected to describe the registration happens with new key?

Reply:
Okay.